crt_portal/cts_forms/views.py

```diff
@@ -1993,12 +1997,25 @@ def get_context_data(self, form, **kwargs):
             'stage_link': True,
             'submit_button': True,
             'form_novalidate': True,
+            'pro_form_attachments': attachments,
         })
 
         return context
 
     def done(self, form_list, form_dict, **kwargs):
-        data, report = save_form(self.get_all_cleaned_data())
+        cleaned_data = self.get_all_cleaned_data()
+        cleaned_data.pop('pro_form_attachment')
+        data, report = save_form(cleaned_data)
+        pro_form_attachments = self.request.POST.get('pro_form_attachment', []).split(',')[:-1]
+        for id in pro_form_attachments:
+            attachment = get_object_or_404(ProformAttachment, pk=int(id))
+            report_attachment = ReportAttachment.objects.create(
+                user=self.request.user,
+                file=attachment.file,
+                filename=attachment.filename,
+                report=report)
+            report_attachment.save()
+            attachment.delete()
```

Comment:
consider switching to a [bulk_create](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#bulk-create) to avoid issues with partially committed transactions

(if something goes wrong - a network error, etc - during the creation of these, it may leave partially created / deleted attachments in the database)